drivers/bridge/bridge.go

```diff
@@ -876,6 +877,55 @@ func setHairpinMode(nlh *netlink.Handle, link netlink.Link, enable bool) error {
 	return nil
 }
 
+func waitForIfaces(ifaceNames []string, timeout time.Duration, trigger func() error) error {
+	// Map interface names to their readiness states (false by default)
+	ifaceStates := make(map[string]bool)
+	allRunning := func() bool {
+		for _, ifaceState := range ifaceStates {
+			if !ifaceState {
+				return false
+			}
+		}
+		return true
+	}
+	for _, ifaceName := range ifaceNames {
+		ifaceStates[ifaceName] = false
+	}
+
+	// Start watching for link changes
+	notifyLink := make(chan netlink.LinkUpdate)
+	notifyDone := make(chan struct{})
+	netlink.LinkSubscribe(notifyLink, notifyDone)
+	defer close(notifyDone)
+
+	// Run the given function to trigger link changes
+	if err := trigger(); err != nil {
+		return err
+	}
+
+	// Do not return until the given interfaces are running, or timeout
+	timer := time.After(timeout)
+	for {
+		select {
+		case update := <-notifyLink:
+			if update.Header.Type == syscall.RTM_NEWLINK &&
+				update.IfInfomsg.Flags&syscall.IFF_RUNNING != 0 {
```

Comment:
I am testing on a 4.4 kernel your changes in a small go program which creates a veth link.
I can confirm the veth running state is notified as up only after both veth sides' links are brought up (which I manually do separately). 

You can also manually verify it with `ip link add v0 type veth peer name v1` and `ip link set v0 up`, `ip link set v1 up` and issue a `ifconfig v0/v1` in between. You will see `RUNNING` in the o/p only after both v0 and v1 links are brought up.

Not sure, but also in CI pretty much all tests which create a veth pair are timing out.